### PR TITLE
Add PVs to Quicklab cluster via NFS

### LINKED ISSUE #37: Quicklab and persistent storage
Currently Openshift4 in Quicklab is lacking persistent storage - no PVs available on the cluster. Let's hope Quicklab team can help us out with that:

https://redhat.service-now.com/surl.do?n=INC1446454

### PR DESCRIPTION
**Pull request by tumido**
Let's use NFS to resolve #37 on our own.

The ansible script bellow allows us to create NFS shares on the Openshift host and expose them to the Openshift itself.

Things to solve:
- Parse out node info from kubernetes and secure the shares to be mountable from Openshift only (currently they are publicly accessible, see `/etc/exports`)
- Confirm the total requested size can really be available on the target host.

### CODE REVIEW

**Comment by martinpovolny** [comment]
Let me test this tomorrow morning.

**Comment by martinpovolny** [comment]
Interesting warning while logging in `load pubkey "/home/martin/.ssh/quicklab.key": invalid format` . But the key loads, is used and works.

**update**:

This seems to be a feature?/bug in ssh: https://bbs.archlinux.org/viewtopic.php?pid=1914847 that is going to get fixed

```
$ rpm -qf $(which ssh )
openssh-clients-8.3p1-3.fc32.x86_6
```

File: docs/downstream/on-cluster-persistent-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+   ```txt
+    /mnt/nfs/A node1(rw) node2(rw) ...
+    ...
+   ```
+
+5. Allow NFS in firewall
+
+   ```sh
+   firewall-cmd --permanent --add-service moun
```
Comment (martinpovolny) [inline docs/downstream/on-cluster-persistent-storage/README.md]: s/moun/mountd

File: docs/downstream/on-cluster-persistent-storage/README.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+   firewall-cmd --permanent --add-service rpc-bind
+   firewall-cmd --permanent --add-service nfs
+   firewall-cmd --reload
+   ```
+
+6. Start and enable NFS service
+
+   ```sh
+   systemctl enable --now nfs-serv
```
Comment (martinpovolny) [inline docs/downstream/on-cluster-persistent-storage/README.md]: s/nfs-serv/nfs-server

**Comment by martinpovolny** [comment]
The manual steps work for me:

pvs-2020-10-05_20-13


Will retry with the ansible playbook next.

**Comment by tumido** [comment]
There were few other issues with the manual setup - it was missing `chown` and `chmod` on the NFS shared directories. Fixed now. I'm still working on the Ansible playbook - shares are not secured yet. I'll remove it from WIP once it's fixed.

File: docs/downstream/on-cluster-persistent-storage/README.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+Bare Openshift cluster installations, like for example Quicklab's Openshift 4 UPI clusters may lack persistent volume setup. This guide will help you set it up.
+
+Please verify that your cluster really lacks `pv`s:
+
+1. Login as a cluster admin
+2. Lookup available `PersistentVolume` resources:
+
+   ```bash
+   \$ oc get pv
```
Comment (martinpovolny) [inline docs/downstream/on-cluster-persistent-storage/README.md]: extra "\\" here?

**Comment by martinpovolny** [comment]
I wonder if a paragraph with prerequisites should be added to the doc (or if it should be considered "common sense"). Something like:

```
ansible-galaxy collection install ansible.posix
ansible-galaxy collection install community.kubernetes
pip install --user openshift pyyaml kubernetes
```

Then `oc login ...` to your cluster and then run the playbook.

**Review by martinpovolny — APPROVED** [review]

**Comment by tumido** [comment]
> I wonder if a paragraph with prerequisites should be added to the doc (or if it should be considered "common sense"). Something like:
> 
> ```
> ansible-galaxy collection install ansible.posix
> ansible-galaxy collection install community.kubernetes
> pip install --user openshift pyyaml kubernetes
> ```
> 
> Then `oc login ...` to your cluster and then run the playbook.

@martinpovolny such paragraph is there already, it's just missing the `pip install` + `oc login` lines... I'll add the `pip install` at least and a mention to login to the cluster...

**Comment by martinpovolny** [comment]
> @martinpovolny such paragraph is there already, it's just missing the `pip install` + `oc login` lines... I'll add the `pip install` at least and a mention to login to the cluster...

Oh, sorry then. It probably was not in an earlier version. I missed it today.

**Comment by tumido** [comment]
@martinpovolny  done,

In the [`pip install` setup step](https://github.com/operate-first/continuous-deployment/pull/49/files#diff-84d21ac751762fcffeca5455c2907fdaR117), I'm explicitly installing `openshift` only, since `kubernetes` and `pyyaml` are both its dependencies.

I've also adjusted names of the playbook task so they fit better + remove my hostname from the sample env file. Now user is forced to change it, and doesn't accidentally use my instance. :smile:

**Comment by tumido** [comment]
let's test this. My hacktoberfest is already complete, but it'll still show up in my profile :smirk:  

/label hacktoberfest-accepted

**Pull request merged**